tests/table/test_select.py

```diff
@@ -1237,3 +1259,53 @@ def test_order_by_raw(self):
                 {"name": "Rustaceans"},
             ],
         )
+
+
+class Color(Table):
+    a_color = Varchar()
+    b_color = Varchar()
+
+
+class TestDistinctOn(TestCase):
+    def setUp(self):
+        Color.create_table().run_sync()
+
+    def tearDown(self):
+        Color.alter().drop_table().run_sync()
+
+    @engines_only("postgres", "cockroach")
```

Comment:
Be careful with this. `DISTINCT ON` works in CockroachDB, but I didn't know how to write tests for Cockroach because there is no sequential primary key and the order of result is different than in Postgres.

Reply:
This looks good to me. Much cleaner than my roughshod take when I first realized it wasn't supported (and done so quickly too!) I will say that I'd prefer if the library did the order_by validation, makes things really clear rather than letting asyncpg or the database tell me something. Sorry I wasn't more help thank you both for your efforts!

Reply:
@williamflaherty Thanks - will see if I we can add some validation.

Reply:
I've added validation now.